Src/FluentAssertions/Types/TypeAssertions.cs

```diff
@@ -72,6 +72,45 @@ public new AndConstraint<TypeAssertions> BeAssignableTo<T>(string because = "",
             return BeAssignableTo(typeof(T), because, becauseArgs);
         }
 
+        bool CheckIfAssignableToOpenGeneric(Type definition)
+        {
+            // The CLR type system does not consider anything to be assignable to an open generic type.
+            // For the purposes of test assertions, the user probably means that the subject type is
+            // assignable to any generic type based on the given generic type definition.
+
+            if (definition.GetTypeInfo().IsInterface)
+            {
+                // check subject and its interfaces against definition
+                TypeInfo subjectInfo = Subject.GetTypeInfo();
+                if (subjectInfo.IsInterface && subjectInfo.IsGenericType &&
+                    subjectInfo.GetGenericTypeDefinition().IsSameOrEqualTo(definition))
+                {
+                    return true;
```

Comment:
🙃 I personally find all these intermediate `return` statements a more difficult to read.

Reply:
Is this better now?